test/test_autograd.py

```diff
@@ -1591,6 +1591,25 @@ def test_inplace_view_of_view(self):
         x.sum().backward()
         self.assertEqual(root.grad.data.tolist(), [[1, 2], [1, 1]])
 
+    def test_inplace_view_volatile(self):
+        # an in-place operation on a view that makes the view volatile should
+        # make the base volatile too
+        base = Variable(torch.randn(2, 2))
+        view = base.narrow(0, 0, 1)
+        view.add_(Variable(torch.randn(1, 2), volatile=True))
+        self.assertTrue(view.volatile)
+        self.assertTrue(base.volatile)
+
+    def test_inplace_base_volatile(self):
+        # an in-place operation on a base that makes the base volatile should
+        # trigger a consistency exception if the view is used in a differentiable
+        # op
+        x = Variable(torch.randn(2, 2), requires_grad=True)
+        base = Variable(torch.randn(2, 2))
+        view = base.narrow(0, 0, 1)
+        base.add_(Variable(torch.randn(1, 2), volatile=True))
+        self.assertRaisesRegex(RuntimeError, 'is_volatile', lambda: view + x)
```

Comment:
Same thing should apply when you apply the volatile operation to the view - it should fail immediately. Can you first add a check before `base.add_` that this happens, and only then make sure that we have consistency checks for regular ops if base was modified.

Reply:
We already test a similar case in test_inplace_view_flags (when base.requires_grad doesn't match the view.) If you in-place added a volatile to the view here, it would succeed and make the base volatile. That's what we test above in test_inplace_view_volatile